### PR TITLE
[bitnami/moodle] Always set `$CFG->wwwroot` to HTTPS if `MOODLE_SSLPROXY` set

### LINKED ISSUE #6535: [bitnami/moodle] Provide a way to disable dynamic scheme in `$CFG->wwwroot`
### Name and Version

bitnami/moodle:4.0.4-debian-11-r2

### What is the problem this feature will solve?

When I enable `$CFG->sslproxy` in `config.php`, I get the following error when trying to access Moodle through my TLS reverse proxy:
"Must use https address in wwwroot when ssl proxy enabled!"

### What is the feature you are proposing to solve the problem?

The issue is caused by this logic in the generated `config.php` that dynamically sets the scheme based on the request:
```
if (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] == 'on') {
  $CFG->wwwroot   = 'https://' . $_SERVER['HTTP_HOST'];
} else {
  $CFG->wwwroot   = 'http://' . $_SERVER['HTTP_HOST'];
}
```
My proposal is to either provide an environment variable that could set `$CFG->wwwroot` statically or to automatically disable this logic when `MOODLE_SSLPROXY` is set (note #6534) and replace it with just:
```
$CFG->wwwroot   = 'https://' . $_SERVER['HTTP_HOST'];
```
Note that `setuplib.php` already does this when `$CFG->sslproxy` is set:
```
        $rurl['scheme'] = 'https'; // make moodle believe it runs on https, squid or something else it doing it
        $_SERVER['HTTPS'] = 'on'; // Override $_SERVER to help external libraries with their HTTPS detection.
        $_SERVER['SERVER_PORT'] = 443; // Assume default ssl port for the proxy.
```
But that only happens later in the process, after `$CFG->wwwroot` has already been set.

### What alternatives have you considered?

For now, I edit `config.php` after it has been generated.

### PR DESCRIPTION
**Pull request by vlk-charles**
### Description of the change

When `$CFG->sslproxy` is set (set by `MOODLE_SSLPROXY`, although see #6534) , `$CFG->wwwroot` is required to start with `https`, otherwise Moodle refuses to serve any content except an error message. Previously, the logic generated by the `moodle_configure_wwwroot` function in `libmoodle.sh` used to determine the scheme of `$CFG->wwwroot` dynamically at run time based on the request.

This pull request enhances the behavior of `moodle_configure_wwwroot` to disable this logic when the `MOODLE_SSLPROXY` environment variable is set. In such case, `$CFG->wwwroot` is set statically to HTTPS.

### Benefits

`/bitnami/moodle/config.php` does not need to be fixed manually after creating a new container.

### Applicable issues

  - fixes #6535

### Additional information

Tested on `bitnami/moodle:4.0.4-debian-11-r2`.

### CODE REVIEW

File: bitnami/moodle/4/debian-11/rootfs/opt/bitnami/scripts/libmoodle.sh [hunk 1]
```diff
@@ -388,14 +388,20 @@ moodle_configure_wwwroot() {
     # sed replacement notes:
     # - The ampersand ('&') is escaped due to sed replacing any non-escaped ampersand characters with the matched string
     # - For the replacement text to be multi-line, an \ needs to be specified to escape the newline character
-    local -r conf_to_replace="if (empty(\$_SERVER['HTTP_HOST'])) {\\
+    local conf_to_replace="if (empty(\$_SERVER['HTTP_HOST'])) {\\
   \$_SERVER['HTTP_HOST'] = '127.0.0.1:${http_port}';\\
-}\\
+}"
```
Comment (rafariossaa) [inline bitnami/moodle/4/debian-11/rootfs/opt/bitnami/scripts/libmoodle.sh:393]: I think should be 
```
}"\\
```
in order to continue the code block.

Comment (vlk-charles) [reply bitnami/moodle/4/debian-11/rootfs/opt/bitnami/scripts/libmoodle.sh:393]: You don't need the backslashes outside the string. The `"` ends the string. The backslashes are for escaping LF for `sed` within the string.

**Comment by vlk-charles** [comment]
`conf_to_replace` is now set in two steps. It is first set to the first part, that is always the same. The second part is chosen depending on whether `MOODLE_SSLPROXY` is set and appended. The appended part always starts with an escaped LF.

**Review by rafariossaa — APPROVED** [review]
LGTM

**Comment by vlk-charles** [comment]
So the changes got merged into the `main` branch. `4.0.4-debian-11-r8` was built after the merge but the commit for it (ba3a5d3) removed my changes. I don't know how the bot works. Is this normal? Will the change reappear later?

**Comment by fmulero** [comment]
Your changes are coming back https://github.com/bitnami/containers/pull/8223/files. Sorry for the inconvenience

**Comment by rafariossaa** [comment]
Hi,
  I am sorry, now it is fixed and published.

**Comment by vlk-charles** [comment]
OK. No problem. I can see the updated script in `4.0.4-debian-11-r9`.

**Pull request merged**